### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by gillins**
This PR adds Windows support. 

It currently skips:
* `icu` (currently only available for Python 3.4)
* `xz` (Windows support not yet submitted to conda-forge but looks like only Python 3.5 will be supported).  

Is this ok for now?

### CODE REVIEW

**Comment by gillins** [comment]
Sorry @msarahan only just noticed your similar PR https://github.com/conda-forge/libxml2-feedstock/pull/7. I'm happy to take your lead on this. Now we have `libiconv` going I was hoping to just get a minimum build of this so we can start building `libspatialite`.

**Comment by msarahan** [comment]
I'm happy to close my PR.  If there's anything valuable there to scavenge, please pull it in here.  Otherwise, I have too much to do, and don't want to be blocking progress on this.

**Comment by gillins** [comment]
Seems that it is only the Python 2.7 tests on Windows that are failing. When I try on my machine I get a R6034 runtime error when starting xmllint.exe. Looking at xlmlint.exe with Dependency Walker shows that the MSVCR90.DLL that is linked by xmllint.exe and libxml2.dll is the Conda one. But the MSVCR90.DLL that iconv.dll and zlib.dll link in lives under C:\windows\winsxs.

@patricksnape I've never understood the winsxs stuff. Is there an easy solution?

**Comment by msarahan** [comment]
That might indicate that two different service packs of vs 2008 are in use,
though if everything is being built on appveyor, I don't see how that would
happen.

Can you look at each dll being looked up and find the exact version?

**Comment by patricksnape** [comment]
As @msarahan said - can we check the DLL versions?

**Comment by gillins** [comment]
Versions of MSVCR90.DLL:

`zlib` and `iconv`: 9.00.30729.4940
`libxml2` and xmllint: 9.00.30729.1

`zlib` and `iconv` where built on AppVeyor, but `libxml2` on my machine so there could be a mismatch there with service packs, but then again the same thing happens with building `libxml2` on AppVeyor...

The thing that bothers me is why Conda is grabbing DLL's from `c:\windows\winsxs` at all. Is this a problem?

**Comment by msarahan** [comment]
It depends on whether the DLL's you're building are set up to use
manifests.  If they are, and they specify a particular version in that
manifest, then Windows will go look in WinSxS if it doesn't find things
sooner.

I'm mystified how that happens with all builds done on AppVeyor, though.

**Comment by gillins** [comment]
Both `libiconv` and `zlib` use `cmake` for building which seems to insert manifest information automatically (example from `iconv.dll` below). Doesn't seem to specify a version though...

`libxml2` has its own build system hence the difference. If `cmake` is a problem then we will have to potentially change a large number of projects. BTW I couldn't easily see how to turn this off in `cmake`.

I would note that it is only MSVC 2008 (and Python 2.7) that use manifests - later versions of VC abandoned it. So it does make sense this is a problem.

Should `conda-build` remove manifest information automatically if this is the problem?

```
<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
<assembly xmlns='urn:schemas-microsoft-com:asm.v1' manifestVersion='1.0'>
  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
    <security>
      <requestedPrivileges>
        <requestedExecutionLevel level='asInvoker' uiAccess='false' />
      </requestedPrivileges>
    </security>
  </trustInfo>
</assembly>
```

**Comment by gillins** [comment]
See down the end of this post for how to disable the manifest in `cmake`: https://blog.kitware.com/deploying-on-windows-with-dll-manifest-issues/
Seems to work by changing the linker flags. I'm hoping we don't have to do this for every `cmake` project we have.

**Comment by patricksnape** [comment]
Yuck, great find @gillins. Perhaps this should be part of our advice then?

**Comment by gillins** [comment]
I guess so. Either that or make a change to conda-build to remove the manifest? I don't know what would be involved with that.

**Comment by gillins** [comment]
OK I've looked into this a bit more and find that is still fails when it is built with `iconv=no icu=no zlib=no lzma=no python=no ftp=no http=no` - ie no external libraries.

However, it will run in the directory it was built in! If you copy the files elsewhere they will fail. They do not appear to have an embedded manifest. Is there any other weird Windows 'feature' I need to be aware of here?

**Comment by msarahan** [comment]
Not that I know of, but I probably have more to learn. @mingwandroid knows
the most of anyone I know.

**Comment by gillins** [comment]
So I found if I enabled the optional manifest mechanism for the VS2008 build worked. I don't know why.

If anyone can shed any light on this it would be appreciated!

The manifest mechanism slightly worries me since it is possible for an application to a request a version of the run time libs that isn't distributed with Conda. If this version is not on the user machine also I'm not sure what would happen.

**Comment by mingwandroid** [comment]
I'm not actually too knowledgable about Windows manifests, on `MSYS2`, we use a standard one for all the i686 executables and don't use any for x86_64, but that's as much as I know I'm afraid. WinSxS does seem deprecated and the UCRT method a better approach.

**Comment by gillins** [comment]
Sounds sensible @mingwandroid. Building with MSVC has certainly added much complication that isn't easily understood...

**Comment by ocefpaf** [comment]
@gillins thanks! Is this good to merge?

**Comment by gillins** [comment]
Yes go for it :+1:

**Comment by msarahan** [comment]
How about patching the manifest to match the vs runtime version that we
ship?

**Comment by gillins** [comment]
> How about patching the manifest to match the vs runtime version that we ship?

You mean across the whole of Conda? Good point. Or remove them entirely (I have some code to do this I can post). Unfortunately I don't know enough about this, but I feel that Manifests are something likely to trip us up in future...

**Comment by ocefpaf** [comment]
> Yes go for it :+1: 


Want to add yourself as a maintainer before I merge?

**Comment by gillins** [comment]
Done - thanks.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -51,3 +51,4 @@ extra:
     recipe-maintainers:
         - ocefpaf
         - jakirkham
+        - gillins
```
Comment (ocefpaf) [inline recipe/meta.yaml]: :+1:

**Comment by gillins** [comment]
Attached is C code for deleting manifests. Currently some packages have them, and others don't. Some like this one won't build without them (at least for VS2008). It's a bit of a mess...

[deletemanifest.c.txt](https://github.com/conda-forge/libxml2-feedstock/files/323287/deletemanifest.c.txt)

**Pull request merged**